src/OrchardCore/OrchardCore.Data.YesSql/OrchardCoreBuilderExtensions.cs

```diff
@@ -83,7 +84,9 @@ public static OrchardCoreBuilder AddDataAccess(this OrchardCoreBuilder builder)
                             var databaseFolder = SqliteHelper.GetDatabaseFolder(shellOptions, shellSettings.Name);
                             Directory.CreateDirectory(databaseFolder);
 
-                            var connectionString = SqliteHelper.GetConnectionString(sqliteOptions, databaseFolder, shellSettings["DatabaseName"]);
+                            // Only allow creating a file DB when a tenant is in the Initializing state
+                            var connectionString = SqliteHelper.GetConnectionString(sqliteOptions, databaseFolder, shellSettings["DatabaseName"], shellSettings.State == TenantState.Initializing);
```

Comment:
Maybe here we can add another helper that would accept `ShellSettings` so this was the logic of creating or not along with the name will be dealt with using the helper.

Reply:
Yeah this could be simplified by passing the entire ShellSettings.

Reply:
Yeah well that why these are just helpers. I think adding a new helper will clean things up. But it does not matter. These helpers are not widley used

Reply:
I second this, let's move the whole logic to the helper. There are too many parameters now too.

Reply:
Ok will create a helper method with and without ShellSettings then. It may be usefull for some.